### PR TITLE
Refactor geo location platform

### PR DESCRIPTION
**Pull request by bieniu**

### CODE REVIEW

File: custom_components/blitzortung/geo_location.py [hunk 1]
```diff
@@ -1,39 +1,46 @@
...
+async def async_setup_entry(
+    hass, config_entry: BlitzortungConfigEntry, async_add_entities
+):
     coordinator = config_entry.runtime_data
     if not coordinator.max_tracked_lightnings:
         return
 
+    # We don't want geo_location entities from the previous Home Assistant run to remain
+    # in the registry and have an unavailable state, so we remove all geo_location 
+    # entities.
```
Comment (mrk-its) [inline custom_components/blitzortung/geo_location.py:36]: :100:

File: custom_components/blitzortung/geo_location.py [hunk 2]
```diff
@@ -139,75 +150,33 @@ class BlitzortungEvent(GeolocationEvent):
...
-    def should_poll(self):
-        """Disable polling."""
-        return False
-
-    @property
-    def unit_of_measurement(self):
-        """Return the unit of measurement."""
-        return self._unit_of_measurement
+        self.entity_id = f"geo_location.lightning_strike_{self._strike_id}"
+        self._attr_name = "Lightning strike"
```
Comment (mrk-its) [inline custom_components/blitzortung/geo_location.py:160]: Nice, didn't know this trick with `_attr` prefix.

**Review by mrk-its — APPROVED** [review]

**Comment by mrk-its** [comment]
@bieniu After merging #129 some conflicts needs to be resolved

**Comment by bieniu** [comment]
> After merging #129 some conflicts needs to be resolved

Done

File: custom_components/blitzortung/geo_location.py (hunk 2 above)
Comment (bieniu) [reply custom_components/blitzortung/geo_location.py:160]: Yeah this is very useful.

**Pull request merged**